services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/FlowHelper.groovy

```diff
@@ -131,13 +145,17 @@ class FlowHelper {
      */
     private FlowEndpointPayload getFlowEndpoint(Switch sw, List<Integer> allowedPorts,
                                                 boolean useTraffgenPorts = true) {
-        def port = allowedPorts[random.nextInt(allowedPorts.size())]
+        allowedPorts -= getSwitchPortsMap()[sw.dpId]
```

Comment:
It'd look better to just use variable directly instead of getter method

Reply:
Unfortunately, it is impossible. We need to init a map where the keys are switch IDs and values are ports occupied by flows. It is not impossible to do at L41 because topology is still null and we fail at this point. So this workaround allows us to avoid failures.